### DISCUSSION TITLE
MKS UI for FlashForge ?

### DISCUSSION THREAD

**Discussion by moonglow** [moonglow/FlashForge_Marlin#28] [Ideas]
I think it looks pretty and more user friendly ( it also has builtin multi language support and model preview mode and UI skins can be customized by user without firmware compilation ), but it needs to add an external cheap SPI FLASH module ( it can be easily connected to WIFI socket ). New FlashForge also can handle MKS WIFI module ( it based on cheapest possible WIFI module EPS8266 ). Motherboard WIFI socket pinout: [LINK](https://github.com/moonglow/FlashForge_Marlin/wiki/FlashForge-WIFI-module-pinout). I'm waiting for modules, to do some tests ( now it runned with emulated flash memory on SD card ,but it works verrry slowly )

----
Some screenshots of MKS UI:
![IMG_20210513_143119](https://user-images.githubusercontent.com/1655387/118150967-ce1c2600-b41b-11eb-9a8b-68b55ce09c17.jpg)
![IMG_20210513_161749](https://user-images.githubusercontent.com/1655387/118150984-d1afad00-b41b-11eb-8966-c8bf51e2aab3.jpg)

**Comment by tckb** [comment]
I really like the MKSUI but problem is this custom mods that has to be fixed in.

**Comment by moonglow** [comment]
@tckb 
Hm... can you provide me more information ? This is Marlin MKS UI from 2.0.8

**Comment by tckb** [comment]
@moonglow  I haven't upgraded to 2.0.8 (lack to time) but as you mentioned in the description, it needs extra components to work with Stock FF board isn't it?

**Comment by moonglow** [comment]
@tckb yep ,but this is easy to install solution you no need to solder it to motherboard just connect it with DuPont wires and that's all folks :)  you can switch back easily all extension boards connected to WIFI socket. This is optional solution, user can choose CLASSIC_UI MARLIN_UI or MKS_UI. As example we can install TMC2209 stepper motor drivers to our original board to make it silent and get better printing quality, this is also changes to original board, but it also optional solution.

![spi_flash](https://user-images.githubusercontent.com/1655387/118239703-cc497580-b4a2-11eb-8260-ef712de009a2.png)

**Comment by zerog2k** [comment]
@moonglow I know this is an old thread, but curious about the MKS UI assets (image/glyph library). Is this not something that can be stored on the internal sd card instead of spi flash?
(I have some experience customizing & porting Marlin, but only to text UI)

edit: I guess mks_ui assumes assets are stored on spi flash, from looking at 
https://github.com/moonglow/FlashForge_Marlin/blob/marlin_2.0.x/Marlin/Marlin/src/lcd/extui/mks_ui/SPI_TFT.cpp
(I guess I'll need to study this more...)

**Reply by moonglow** [reply]
@zerog2k Hi!

>Is this not something that can be stored on the internal sd card instead of spi flash?

This is was my first test build... i used external SDCard to store and access to UI assets, so i simulates SPI flash as binary file, but random reads over FatFs it slows as hell... and MKS UI is not really useful, but it looking pretty nice yeah
